automation/orchestrator.go

```diff
@@ -43,15 +45,22 @@ type ApplyFactory func(ctx context.Context) Apply
 
 // Orchestrator manages automation in qri
 type Orchestrator struct {
+	// TODO(ramfox): this runLock is the current shim to ensure only one workflow runs at a time
+	// we should probably have a run queue subsystem that ensure the orchestrator is running
+	// the workflows in the expected order, running only as many at once as configured, and
+	// allows communication back to the user about where they are in the run queue, allows for
+	// cancelling runs that haven't happened yet
+	runLock      sync.Mutex
```

Comment:
So there was a bug in the code caught by the orchestrator integration test. Previously, we discussed that as a simple way to keep workflows from not running simultaneously, we should have the handlers for the `event.ETTriggerEvent` be blocking.

However, this meant that we couldn't use the bus for any functionality during the run... which we need to do for the run store, since we need to collect transform events off the bus in order to keep the run state up to date.

Instead, using a mutex lock for now to ensure runs are sequential, and changed the `handleTrigger` event handler to be non-blocking.

Reply:
Very interesting. I agree that the best solution would be a queue with a lock, but this is a fine stop-gap for now.